### DISCUSSION TITLE
Invite email link

### DISCUSSION THREAD

**Discussion by daniellin215** [nextauthjs/next-auth#4208] [Help]
### Description 📓

Hi, is there a way to send invite email to new users, and upon them signing in, we also pass some parameters such as the permissions set by the user who invited them. Furthermore, would this only work with the EmailProvider or can this also somehow work with the GoogleProvider?

### How to reproduce ☕️

See description

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by volere** [comment]
I was wondering about this too, I was looking at the [the email provider](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/email.ts). I am wondering if it would work to use this code outside of [...nextauth].js ?

**Comment by DinerIsmail** [comment]
@daniellin215 @volere have you had any luck implementing an invite feature? I was just wondering how to achieve that as well.

**Comment by magicspon** [comment]
@daniellin215 @volere 
I've managed to hack the "email" sign up process to work with invites... 
might need a bit of work:

https://gist.github.com/magicspon/614a780d567a08334831347338a3d262

the key is to overwrite the `sendVerificationRequest` function, and use an api route as the callback url.
